### ISSUE TITLE
Better Prop Check Failure Reporting

### ISSUE THREAD

**Issue by expede** [expede/type_class#19]
Currently, prop checks fail with minimal context. It would be good to have more context on which specific values were the cause.

**Comment by expede** [comment]
@paulruescher, you've expressed interest in doing some work on the Witchcraft suite. Would you like to give this a look? Totally happy to walk you through how the library works 👍

**Comment by OvermindDL1** [comment]
Since [StreamData](https://github.com/whatyouhide/stream_data) has come out by one of the primary Elixir Devs (and thus being used by some of the biggest Elixir projects) have you thought about using it as the property testing library here?  It would be a more unified interface for defining types.  It's not as powerful as PropEr or so, but it is simple and fast with good error reporting and decent reduction.  :-)

**Comment by paulruescher** [comment]
> you've expressed interest in doing some work on the Witchcraft suite. Would you like to give this a look

Yeah, I'd be down to tackle this. Not sure _when_ though 😎I'm definitely outta commish this week, but I'll see what I can do next week.
